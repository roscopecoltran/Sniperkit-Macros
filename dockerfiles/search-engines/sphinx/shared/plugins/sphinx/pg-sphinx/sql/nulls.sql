SELECT * FROM sphinx_select(NULL, 'database', 'user_id = 12', 'weight DESC, id DESC', 0, 100, NULL);

SELECT * FROM sphinx_select('test_index', NULL, 'user_id = 12', 'weight DESC, id DESC', 0, 100, NULL);

SELECT * FROM sphinx_select('test_index', 'database', NULL, 'weight DESC, id DESC', 0, 100, NULL);

SELECT * FROM sphinx_select('test_index', 'database', 'user_id = 12', NULL, 0, 100, NULL);

SELECT * FROM sphinx_select('test_index', 'database', 'user_id = 12', 'weight DESC, id DESC', NULL, 100, NULL);

SELECT * FROM sphinx_select('test_index', 'database', 'user_id = 12', 'weight DESC, id DESC', 0, NULL, NULL);


SELECT sphinx_snippet(NULL, 'database', 'Text about databases', '<mark>', '</mark>');

SELECT sphinx_snippet('test_index', NULL, 'Text about databases', '<mark>', '</mark>');

SELECT sphinx_snippet('test_index', 'database', NULL, '<mark>', '</mark>');

SELECT sphinx_snippet('test_index', 'database', 'Text about databases', NULL, '</mark>');

SELECT sphinx_snippet('test_index', 'database', 'Text about databases', '<mark>', NULL);


SELECT sphinx_snippet_options(NULL, 'database', 'Text about databases',
  ARRAY['before_match', '<mark>',
        'after_match', '</mark>',
        'limit', '100',
        'allow_empty', 'y']);

SELECT sphinx_snippet_options('test_index', NULL, 'Text about databases',
  ARRAY['before_match', '<mark>',
        'after_match', '</mark>',
        'limit', '100',
        'allow_empty', 'y']);


SELECT sphinx_snippet_options('test_index', 'database', NULL,
  ARRAY['before_match', '<mark>',
        'after_match', '</mark>',
        'limit', '100',
        'allow_empty', 'y']);

SELECT sphinx_snippet_options('test_index', 'database', 'Text about databases', NULL);

SELECT sphinx_snippet_options('test_index', 'database', 'Text about databases',
  ARRAY[]::text[]);

SELECT sphinx_snippet_options('test_index', 'database', 'Text about databases',
  ARRAY['before_match']);

