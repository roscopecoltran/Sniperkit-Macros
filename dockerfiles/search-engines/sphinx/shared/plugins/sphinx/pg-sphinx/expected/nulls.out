SELECT * FROM sphinx_select(NULL, 'database', 'user_id = 12', 'weight DESC, id DESC', 0, 100, NULL);
ERROR:  'index' parameter has to be a valid name of search index but NULL was given.
SELECT * FROM sphinx_select('test_index', NULL, 'user_id = 12', 'weight DESC, id DESC', 0, 100, NULL);
ERROR:  'query' parameter has to be a search query but NULL was given.
SELECT * FROM sphinx_select('test_index', 'database', NULL, 'weight DESC, id DESC', 0, 100, NULL);
 id | weight 
----+--------
(0 rows)

SELECT * FROM sphinx_select('test_index', 'database', 'user_id = 12', NULL, 0, 100, NULL);
 id | weight 
----+--------
(0 rows)

SELECT * FROM sphinx_select('test_index', 'database', 'user_id = 12', 'weight DESC, id DESC', NULL, 100, NULL);
 id | weight 
----+--------
(0 rows)

SELECT * FROM sphinx_select('test_index', 'database', 'user_id = 12', 'weight DESC, id DESC', 0, NULL, NULL);
 id | weight 
----+--------
(0 rows)

SELECT sphinx_snippet(NULL, 'database', 'Text about databases', '<mark>', '</mark>');
 sphinx_snippet 
----------------
 
(1 row)

SELECT sphinx_snippet('test_index', NULL, 'Text about databases', '<mark>', '</mark>');
 sphinx_snippet 
----------------
 
(1 row)

SELECT sphinx_snippet('test_index', 'database', NULL, '<mark>', '</mark>');
 sphinx_snippet 
----------------
 
(1 row)

SELECT sphinx_snippet('test_index', 'database', 'Text about databases', NULL, '</mark>');
       sphinx_snippet        
-----------------------------
 Text about databases</mark>
(1 row)

SELECT sphinx_snippet('test_index', 'database', 'Text about databases', '<mark>', NULL);
       sphinx_snippet       
----------------------------
 Text about <mark>databases
(1 row)

SELECT sphinx_snippet_options(NULL, 'database', 'Text about databases',
  ARRAY['before_match', '<mark>',
        'after_match', '</mark>',
        'limit', '100',
        'allow_empty', 'y']);
 sphinx_snippet_options 
------------------------
 
(1 row)

SELECT sphinx_snippet_options('test_index', NULL, 'Text about databases',
  ARRAY['before_match', '<mark>',
        'after_match', '</mark>',
        'limit', '100',
        'allow_empty', 'y']);
 sphinx_snippet_options 
------------------------
 
(1 row)

SELECT sphinx_snippet_options('test_index', 'database', NULL,
  ARRAY['before_match', '<mark>',
        'after_match', '</mark>',
        'limit', '100',
        'allow_empty', 'y']);
 sphinx_snippet_options 
------------------------
 
(1 row)

SELECT sphinx_snippet_options('test_index', 'database', 'Text about databases', NULL);
 sphinx_snippet_options 
------------------------
 
(1 row)

SELECT sphinx_snippet_options('test_index', 'database', 'Text about databases',
  ARRAY[]::text[]);
 sphinx_snippet_options 
------------------------
 
(1 row)

SELECT sphinx_snippet_options('test_index', 'database', 'Text about databases',
  ARRAY['before_match']);
 sphinx_snippet_options 
------------------------
 
(1 row)

